Refactor Background into separate component, change window scroll hook, fixes 79

mike1808 opened the pull request:
* Refactor Background into a separate component to avoid re-rendering the whole app on scroll event
* Change `useWindowScroll` hook to `react-use` as it uses `requestAnimationFrame` which make animation on scroll event more smoother
* Change usage of `background-position-y` to `transform: translateY()` to use the browsers hardware acceleration

nairihar commented:
Hey @mike1808, thanks for the PR.
There is a big space on the bottom. Is it something related to the Image size?

mike1808 commented:
@nairihar sorry for the late response, been busy recently. I fixed that issue, also, I changed the way the style is updated (no React for that).

And, btw, fixed one more bug on mobile devices

Was
![image](https://user-images.githubusercontent.com/3942554/56053375-c6539a00-5d08-11e9-94a8-886b774e79aa.png)

Become
![image](https://user-images.githubusercontent.com/3942554/56053392-d1a6c580-5d08-11e9-9235-a940f7da5cff.png)

mike1808 commented:
You can see FPS changes: open dev tools, press "Ctrl Shift P", find "Show fps" and try to scroll this PR and the original site.

nairihar commented:
@mike1808 Thanks for the changes, Looks Good 👍
